upickle/src/upickle/Api.scala

```diff
@@ -71,50 +88,103 @@ trait Api
   def writeTo[T: Writer](t: T,
                          out: java.io.Writer,
                          indent: Int = -1,
-                         escapeUnicode: Boolean = false): Unit = {
-    transform(t).to(new ujson.Renderer(out, indent = indent, escapeUnicode))
+                         escapeUnicode: Boolean = false,
+                         sortKeys: Boolean = false): Unit = {
+    maybeSortKeysTransform(t, sortKeys, new ujson.Renderer(out, indent = indent, escapeUnicode))
   }
+
+  // Binary Compatibility Stub
```

Comment:
Should be a `@deprecated` annotation with version, so we can better clean up later.

Reply:
> 4\. Binary compatibility stubs have been added to all methods who received a new `sortKeys: Boolean = true` parameter. These stubs were not marked as `@Deprecated`, because in some cases the Scala compiler would preferentially resolve the stub over the method-with-default-value, causing spurious deprecation warnings.

Do you have an example where this happens? This can be most likely worked around by specifying all parameters of the new `def`, including the defaults. Or we could add a `: @nowarn` if we think it's ok from a bincompat perspective.

If the compiler is still using a deprecated `def`, than it's clearly _not_ just a stub to keep backward compatibility.

Reply:
I think the case was something like

```scala
// actual method
def foo(a: Int, b: Int = 0) = ...

// bincompat stub
def foo(a: Int) = foo(a, b = 0)

foo(1) // This calls the bincompat stub
```

The issue is that this would cause warnings in downstream user code, not just within uPickle's own codebase. I don't want to have users have to sprinkle `@nowarn` annotations to silence meaningless deprecation warnings

Reply:
From a vague memory, the hack to make that work was to reduce visibility with some scoped protected/private stuff, which probably works, as these restrictions are (only) compile-time enforced but not at runtime. So code that links to these methods, will be able to use it, but the compiler will not. But I think MiMa will still complain. But I'm not 100% sure anymore. 

We could probable just remove the default argument and go with classic overloading. Maybe, leave a note for when we prepare the next breakage?

Reply:
Didn't manage to get it to work using various combinations of `private`; either the deprecation warning would fire, or mima would complain

For now just left comments for all the binary compatibility stub methods. Should be easy to grep for them later